File: kaprekar.py
numero = "3962"

def algoritmo(numero):
	numprincipal = numero

	cont = 0

	numdiferencia = 0

	while numdiferencia != 6174:
		numero = [str(numero[0]), str(numero[1]), str(numero[2]), str(numero[3])]

		listaCopia = numero[:]
		listAux = []

		numeromenor = sorted(numero)

		for i in range(len(numero)):
			aux = 0
			aux = max(listaCopia)
			listAux.append(aux)
			listaCopia.remove(aux)

		num = ""
		nummenor = ""
		nummayor = ""

		for i in range(len(numero)):
			num = num + numero[i]
			nummenor = nummenor + numeromenor[i]
			nummayor = nummayor + listAux[i]

		num = int(num)
		nummenor = int(nummenor)
		nummayor = int(nummayor)

		numdiferencia = nummayor - nummenor
		numero = str(numdiferencia)

		if len(numero) != 4:
			numero =  "0" + numero

		cont = cont + 1


	print("El número de iteraciones para el número " + numprincipal + " es: " + str(cont))

algoritmo(numero)

File: kaprekar.js
var numero = "3962";

function kaprekar(numero){

  var numero_copia = numero;
  var cont = 0;
  var numdiferencia = 0;
  
  do{
  
    var numero = ["string de numero[0]", "string de numero[1]", "string de numero[2]", "string de numero[3]"];
    
    var listaCopia = "copia de lista numero";
    var listaAux = [];
    
    var numeromenor = numero.sort;
    
    for (var i = 0; i <= numero.length; i++){
      var aux = 0;
      aux = "maximo de listaCopia";
      listAux.add(aux);
      listaCopia.delete(aux);
    }
    
    var num = "";
    var nummenor = "";
    var nummayor = "";
    
    for(var i = 0; i <= numero.length; i++){
      num = num + numero[i];
      nummenor = nummenor + numeromenor[i];
      nummayor = nummayor + listAux[i];
    }
    
    num = num.parseInt;
    nummenor = nummenor.parseInt;
    nummayor = nummayor.parseInt;
    
    numdiferencia = nummayor - nummenor;
    numero = numdiferencia.parseInt;
    
    if(numero.length != 4){
      numero = "0" + numero;
    }
    
    cont = cont + 1;
  }while(numdiferencia != 6174);
  
  console.log("El número de iteraciones para el número " + numero_copia + " es: " + cont.toString);
}
